cmd/prometheus/main.go

```diff
@@ -1083,14 +1090,29 @@ func (s *readyStorage) Set(db *tsdb.DB, startTimeMargin int64) {
 	s.startTimeMargin = startTimeMargin
 }
 
-// get is internal, you should use readyStorage as the front implementation layer.
+// SetStats sets the stats object,
+// which should be available before the DB is finished opening.
+func (s *readyStorage) SetStats(stats *tsdb.DBStats) {
```

Comment:
Do we need this method? Why not passing this through constructor always?

Reply:
or even better, created through constructor?

Reply:
If you mean adding it to the variable definition over here instead of using that function, I'm pretty sure that will work.

https://github.com/prometheus/prometheus/blob/9a19336b29060eb0414e80382b904777a002e91c/cmd/prometheus/main.go#L436-L437

Reply:
Will work? Can we do this? Limiting number of methods is always helpful.

Reply:
Yup!